<?xml version="1.0"?>
<launch>
  <arg name="ns" default="depthai"/>
  <arg name="node_name" default="driver"/>

  <arg name="camera_name" default="bw1097"/>
  <arg name="camera_param_uri" default="package://depthai_ros_driver/params/camera/"
    doc="Location of parameters is {camera_param_uri}/{camera_name}/{stream}.yaml"/>

  <arg name="pipeline" default="depthai_ros_driver/MobilenetSSDPipeline"/>

  <arg name="cmd_file" default="$(find depthai_ros_driver)/resources/depthai.cmd"/>
  <arg name="calibration_file" default="$(find depthai_ros_driver)/resources/calibration/$(arg camera_name)/depthai.calib"/>

  <arg name="blob_file" default="$(find depthai_ros_driver)/resources/mobilenet_ssd.blob"/>
  <arg name="blob_file_config" default="$(find depthai_ros_driver)/resources/mobilenet-ssd.json"/>
  <arg name="openvino_version" default="2" doc="integer corresponding to dai::OpenVINO::Version"/>

  <arg name="use_gdb" default="false"/>
  <arg name="launch_prefix" value="/usr/bin/gdb --args" if="$(arg use_gdb)"/>
  <arg name="launch_prefix" value="" unless="$(arg use_gdb)"/>

  <!--
  Possible streams:
  * 'left' - left mono camera preview
  * 'right' - right mono camera preview
  * 'previewout' - preview of stream sent to the NN
  * 'metaout' - CNN output tensors
  * 'depth' - the raw depth map, disparity converted to real life distance
  * 'disparity' - disparity map, the diaparity between left and right cameras, in pixels
  * 'disparity_color' - disparity map colorized
  * 'meta_d2h' - device metadata stream
  * 'video' - H.264/H.265 encoded color camera frames
  * 'jpegout' - JPEG encoded color camera frames
  * 'object_tracker' - Object tracker results
  -->
  <arg name="stream_list" default="[left, right, metaout, previewout, disparity, disparity_color, depth, video]"/>

  <!-- Launch the Depthai node -->
  <group ns="$(arg ns)">
    <node pkg="depthai_ros_driver" type="gen2_ros_driver" name="$(arg node_name)" output="screen" launch-prefix="$(arg launch_prefix)">
      <param name="camera_name" value="$(arg camera_name)"/>
      <param name="camera_param_uri" value="$(arg camera_param_uri)"/>

      <param name="pipeline" value="$(arg pipeline)"/>

      <param name="cmd_file" value="$(arg cmd_file)"/>
      <param name="calibration_file" value="$(arg calibration_file)"/>
      <param name="blob_file" value="$(arg blob_file)"/>
      <param name="blob_file_config" value="$(arg blob_file_config)"/>
      <param name="openvino_version" value="$(arg openvino_version)"/>

      <rosparam param="stream_list" subst_value="True">$(arg stream_list)</rosparam>
    </node>
  </group>

  <!-- Publish tf -->
  <param name="robot_description" command="$(find xacro)/xacro '$(find depthai_ros_driver)/urdf/$(arg camera_name).urdf.xacro'"/>
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">
    <param name="tf_prefix" value=""/>
  </node>
</launch>
